pkg/azure/actuator.go

```diff
@@ -363,11 +372,19 @@ func (a *Actuator) syncMint(ctx context.Context, cr *minterv1.CredentialsRequest
 
 	spName := azureStatus.ServicePrincipalName
 	if spName == "" {
-		spName, err = generateServicePrincipalName(infraName, cr.Name)
+		spName, err = a.generateServicePrincipalName(infraName, cr.Name)
 		if err != nil {
 			return err
 		}
 		azureStatus.ServicePrincipalName = spName
+
+		// Save status immediately in case we successfully create the AppRegistration and
+		// then fail to update credReq status causing us to leak the created AppRegistration
+		// because we can't find it by name anymore.
+		err = a.updateProviderStatus(ctx, logger, cr, azureStatus)
```

Comment:
Even `updateProviderStatus` can error in which case the created AAD application will be orphaned and new AAD application will be created.